test/routes.spec.js

```diff
@@ -0,0 +1,297 @@
+process.env.NODE_ENV = 'test';
+
+const chai = require('chai');
+const should = chai.should();
+const chaiHttp = require('chai-http');
+const server = require('../server');
+
+const environment = process.env.NODE_ENV || 'test';
+const configuration = require('../knexfile')[environment];
+const database = require('knex')(configuration);
+
+chai.use(chaiHttp);
+
+beforeEach((done) => {
+    database.migrate.latest()
+      .then(() => done())
+      .catch(error => {
+        throw error;
+      });
+  });
+  beforeEach((done) => {
+    database.seed.run()
+      .then(() => done())
+      .catch(error => {
+        throw error;
+      });
+  });
+  afterEach((done) => {
+    database.migrate.rollback()
+    .then(() => done())
+    .catch(error => {
+        throw error;
+      });
+  });
+
+describe('API Routes', () => {
+    it('should return the homepage with text', done => {
+        chai.request(server)
+        .get('/')
+        .end((err, response) => {
+            response.should.have.status(200);
+            response.should.be.html;
+            response.res.text.should.equal('Welcome to Play API');
+            done();
+        });
+    });
+    it('should return a 404 for a route that does not exist', done => {
+        chai.request(server)
+        .get('/sad')
+        .end((err, response) => {
+            response.should.have.status(404);
+            done();
+        });
+    });
+    it('should return favorites', done => {
+        chai.request(server)
+        .get('/api/v1/favorites')
+        .end((err, response) => {
+            response.should.have.status(200);
+            response.should.be.json;
+            response.body.should.be.a('array');
+            response.body.length.should.equal(3);
+            response.body[0].should.have.property('id');
+            response.body[0].id.should.equal(1);
+            response.body[0].should.have.property('song_title');
+            response.body[0].song_title.should.equal('Fight Me');
+            response.body[0].should.have.property('artist_name');
```

Comment:
Since you test this particular album and band in the show, it might have been helpful to look for another band, just in case

Reply:
Good point! Thanks